appcli/configuration_manager.py

```diff
@@ -43,6 +45,33 @@
 METADATA_FILE_NAME = "metadata-configure-apply.json"
 """ Name of the file holding metadata from running a configure (relative to the generated configuration directory) """
 
+
+class FileLoader:
```

Comment:
I think we should just simplify this.
Basically you have a validator concept.

```
file_to_dict_converters = {
    "json": json.load,
    "jsn": json.load,
    "yaml": yaml.safe_load,
    "yml": yaml.safe_load,
}

def __validate_schema(data_file: Path, schema_file: Path):
  converter = file_to_dict_converters[data_file.suffix]
  if converter is None:
    throw etc...

  # Load as you do now
  ...

   # Validate (move the code you have below) 
  data = converter(data_file)
  schema = json.load(schema_file)
  try:
      jsonschema.validate(instance=data, schema=schema)
      logger.debug(
          f"The config file {config_file} matched the provided schema."
      )
  except jsonschema.exceptions.ValidationError as e:
      error_and_exit(f"Validation of {config_file} failed at:\n{e}")
```

So in the below `validate_configuration` you can just do:

```
...
for schema_file in schema_files:
  ...
  __validate_schema(data, schema)
```

That does away with all the loader instances, and means we don't need to for loop through loaders for each file.